src/spec.ts

```diff
@@ -336,6 +361,9 @@ function fieldDefIndex(spec: GenericSpec<GenericUnitSpec<any, any>>, dict: any =
   } else if (isFacetSpec(spec)) {
     accumulate(dict, vlEncoding.fieldDefs(spec.facet));
     fieldDefIndex(spec.spec, dict);
+  } else if (isRepeatSpec(spec)) {
+    accumulate(dict, vlEncoding.fieldDefs(spec.spec));
+    fieldDefIndex(spec.spec, dict);
```

Comment:
Shit!.  This will be wrong and difficult for repeat.  

(Basically we don't really know about this unless we instantiate a model.)